# fishfASE
## Using allele specific expression (ASE) to identify <i>cis-</i> and <i>trans-</i>regulatory divergence contributing to rapidly evolving craniofacial phenotypes.
This is the home for scripts I used to identify gene regulatory mechanisms underlying adaptive trait divergence and reproductive isolation in Cyprinodon pupfish. I used a combination of these scripts for two related publications:

<b>McGirr JA</b> and Martin CH. 2020. Few fixed variants between trophic specialist pupfish species reveal candidate cis-regulatory alleles underlying rapid craniofacial divergence. <i>Molecular Biology and Evolution</i> [early access pdf](https://github.com/joemcgirr/joemcgirr.github.io/blob/master/files/papers/mcgirr_2020.pdf)

<b>McGirr JA</b> and Martin CH. 2020. Ecological divergence in sympatry causes gene misexpression in hybrids. <i>Molecular Ecology</i> [pdf](https://github.com/joemcgirr/joemcgirr.github.io/blob/master/files/papers/mcgirr_2020b.pdf)

Generic command line arguments are outlined in `/genomes/README.md` and `/transcriptomes/README.md` 

Jupyter notebooks in `/genomes` and `/transcriptomes` generate SLURM scripts to run commands.

`/markdown` contains scripts used to analyze output in `/markdown/data_files`

I hope that this repo can be useful to others studying genotype-phenotype relationships in non-model organisms! 

![Image of Pipeline](https://github.com/joemcgirr/fishfASE/blob/master/pipeline.png)

