{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# genomic analyses\n",
    "## generate scripts to:\n",
    "1. trim and align reads\n",
    "2. call snps with gatk 3.8\n",
    "3. calculate fst with vcftools\n",
    "4. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import io\n",
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### San Salvador Island Generalists\n",
    "[\"RHPA1\",\"SPPA1\",\"CRPA3\",\"MRKA1\",\"PIGA3\",\"WDPA1\",\"OSPA4\",\"ME2A2\",\"OSPA7\",\"OYSA1\",\"PIGA1\",\"OSPA6\",\"OSPA5\",\"CRPA1\",\"MERA2\",\"GNYA1\",\"LILA1\",\"ME2A1\",\"GREA1\",\"OSPA1\",\"OSPA9\",\"OYSA2\",\"CLRA1\",\"OSPA11\",\"NLLA1\",\"OSPA13\",\"GREA2\",\"OSPA12\",\"OSPA8\",\"PAIA1\",\"MERA3\",\"OSPA10\"]\n",
    "### San Salvador Island Molluscivores\n",
    "[\"OYSM8\",\"CRPM8\",\"CRPM7\",\"OSPM6\",\"OSPM7\",\"LILM5\",\"OYSM6\",\"OSPM5\",\"CRPM3\",\"CRPM6\",\"MRKM5\",\"OYSM1\",\"CRPM9\",\"CRPM2\",\"LILM4\",\"CRPM11\",\"OSPM11\",\"CRPM10\",\"OSPM3\",\"OSPM9\",\"OYSM3\",\"MRKM3\",\"OYSM2\",\"OSPM2\",\"OYSM5\",\"WDPM2\",\"LILMQ\",\"OSPM4\",\"OSPM10\",\"OSPM8\",\"LILM3\",\"OYSM4\",\"OSPM1\",\"OYSM7\",\"MRKM1\",\"MRKM4\",\"MRKM2\",\"CRPM1\",\"CRPM5\"]\n",
    "### San Salvador Island Scale-eaters\n",
    "\n",
    "[\"ME2P1\",\"CRPP3\",\"CRPP7\",\"OSPP9\",\"LILP4\",\"CRPP8\",\"OSPP1\",\"LILP3\",\"OYSP6\",\"OSPP8\",\"LILPQ\",\"OYSP7\",\"OYSP1\",\"CRPP5\",\"OSPP3\",\"OSPP11\",\"CRPP9\",\"CRPP2\",\"OSPP5\",\"CRPP4\",\"CRPPQ\",\"LILP5\",\"OSPP6\",\"OSPP7\",\"OSPP10\",\"OSPP2\",\"OSPP4\",\"OYSP3\",\"OYSP4\"]\n",
    "### San Salvador Island Small-jawed Scale-eaters\n",
    "\n",
    "[\"OSPS8\",\"OSPS2\",\"GRES4\",\"OYSS3\",\"GREP1\",\"OSPS11\",\"OSPS5\",\"LILS1\",\"GRES3\",\"OYSS5\",\"GREP2\",\"OSPS3\",\"LILS3\",\"LILS4\",\"MERP1\",\"LILS2\",\"OSPS7\",\"OSPS10\",\"OSPS1\",\"OSPS9\",\"OYSS4\",\"OYSS6\",\"MERP2\",\"OYSS9\",\"OSPS6\",\"OSPS4\"]\n",
    "### Caribbean Generalists and Outgroups\n",
    "[\"MEGQ1\",\"GEO2A8\",\"CATA1\",\"CUNP5\",\"BAVA8\",\"ARTA2\",\"GEO2A5\",\"VENA1\",\"CURA21\",\"VENA12\",\"GEO2A9\",\"NCCA11\",\"NCCA4\",\"GEOA2\",\"NCCA2\",\"NCCA5\",\"NCCA12\",\"CUNP7\",\"NCCA1\",\"NCCA15\",\"GEOA10\",\"BAVA11\",\"CUNA6\",\"NCCA9\",\"BAVA2\",\"FCTA1\",\"BAVA4\",\"VENA5\",\"CUNA1\",\"GEOA6\",\"VENA10\",\"MAY1\",\"BAVA6\",\"CUNP4\",\"VENA2\",\"BAVA5\",\"CUNA2\",\"BAVA10\",\"CUNA7\",\"VENA3\",\"GEO2A1\",\"CUNA10\",\"EXUA2\",\"NBIA1\",\"BAVA14\",\"CUNA4\",\"FLSA1\",\"CUNP11\",\"MAFA1\",\"VENA9\",\"CUNA3\",\"VENA7\",\"PWLA1\",\"ETA1\",\"GEOA7\",\"GEO2A6\",\"BAVA42\",\"BAVA7\",\"VENA8\",\"BAVA9\",\"GEO2A7\",\"NCCA8\",\"SIM1\",\"GEOA1\",\"ARTA1\",\"EPLA1\",\"NCCA3\",\"ACKA1\",\"GEOA5\",\"NCCA10\",\"VENA13\",\"GEOA4\",\"CUNA9\",\"KILA1\",\"GEOA11\",\"BAVA13\",\"VENA4\",\"CAIA1\",\"BAVA41\",\"CUNP3\",\"LGIA1\",\"EXUA1\",\"CUNA8\",\"DEAA1\",\"BAVA12\",\"CURA1\",\"CUNP6\",\"CURA2\",\"BONA1\",\"CUNA5\",\"NCCA7\",\"GEO2A3\",\"GEO2A10\",\"GEO2A4\",\"SCLA1\",\"GEO2A2\",\"SALA1\",\"NCCA6\"]\n",
    "fdir = \"/pine/scr/j/m/jmcgirr/pupfish_genomes/Caribbean_pups/\"\n",
    "### San Salvador Island Breeders Used to Generate F1s for RNAseq\n",
    "[\"CRPM1001\",\"CRPP1000\",\"CRPM1000\",\"NCCA1000\",\"CRPP1001\",\"CRPA1000\",\"OSPM1001\",\"CRPA1001\",\"CUNP10.2\",\"OSPA1001\",\"OSPP1000\",\"OSPA1000\",\"CRPA1003\",\"OSPM1000\",\"OSPP1001\"]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "job = 'unzip_san_sal_genomes'\n",
    "\n",
    "#sbatch submission script header\n",
    "script = 'script_' + job + '.sh'\n",
    "outfile = io.open(script,'w', newline='\\n')    \n",
    "outfile.write('#!/bin/bash\\n\\n#SBATCH --job-name='+job+'\\n')\n",
    "outfile.write('#SBATCH --mem=4G\\n')\n",
    "outfile.write('#SBATCH --ntasks=1\\n')\n",
    "outfile.write('#SBATCH --time=24:00:00 ##\\n')\n",
    "outfile.write('#SBATCH -e '+job+'_%A_%a.err ##error file if you want one (will be per job)\\n')\n",
    "outfile.write('#SBATCH --mail-user=jmcgirr@email.unc.edu ##email you when job starts,ends,etc\\n#SBATCH --mail-type=ALL\\n\\n')\n",
    "    \n",
    "#sbatch command\n",
    "outfile.write('gzip -d /pine/scr/j/m/jmcgirr/pupfish_genomes/san_sal/raw_fastq/A/* \\n')\n",
    "outfile.write('gzip -d /pine/scr/j/m/jmcgirr/pupfish_genomes/san_sal/raw_fastq/M/* \\n')\n",
    "outfile.write('gzip -d /pine/scr/j/m/jmcgirr/pupfish_genomes/san_sal/raw_fastq/P/* \\n')\n",
    "outfile.write('gzip -d /pine/scr/j/m/jmcgirr/pupfish_genomes/san_sal/raw_fastq/S/* \\n')\n",
    "\n",
    "#run sbatch submission \n",
    "outfile.write('\\n\\n#command to run it on longleaf is sbatch '+script)\n",
    "outfile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "job = '_trim'\n",
    "# A\n",
    "#infiles = [\"RHPA1\",\"SPPA1\",\"CRPA3\",\"MRKA1\",\"PIGA3\",\"WDPA1\",\"OSPA4\",\"ME2A2\",\"OSPA7\",\"OYSA1\",\"PIGA1\",\"OSPA6\",\"OSPA5\",\"CRPA1\",\"MERA2\",\"GNYA1\",\"LILA1\",\"ME2A1\",\"GREA1\",\"OSPA1\",\"OSPA9\",\"OYSA2\",\"CLRA1\",\"OSPA11\",\"NLLA1\",\"OSPA13\",\"GREA2\",\"OSPA12\",\"OSPA8\",\"PAIA1\",\"MERA3\",\"OSPA10\"]\n",
    "#fdir = \"/pine/scr/j/m/jmcgirr/pupfish_genomes/san_sal/raw_fastq/A/\"\n",
    "# M\n",
    "#infiles = [\"OYSM8\",\"CRPM8\",\"CRPM7\",\"OSPM6\",\"OSPM7\",\"LILM5\",\"OYSM6\",\"OSPM5\",\"CRPM3\",\"CRPM6\",\"MRKM5\",\"OYSM1\",\"CRPM9\",\"CRPM2\",\"LILM4\",\"CRPM11\",\"OSPM11\",\"CRPM10\",\"OSPM3\",\"OSPM9\",\"OYSM3\",\"MRKM3\",\"OYSM2\",\"OSPM2\",\"OYSM5\",\"WDPM2\",\"LILMQ\",\"OSPM4\",\"OSPM10\",\"OSPM8\",\"LILM3\",\"OYSM4\",\"OSPM1\",\"OYSM7\",\"MRKM1\",\"MRKM4\",\"MRKM2\",\"CRPM1\",\"CRPM5\"]\n",
    "#fdir = \"/pine/scr/j/m/jmcgirr/pupfish_genomes/san_sal/raw_fastq/M/\"\n",
    "# P\n",
    "#infiles = [\"ME2P1\",\"CRPP3\",\"CRPP7\",\"OSPP9\",\"LILP4\",\"CRPP8\",\"OSPP1\",\"LILP3\",\"OYSP6\",\"OSPP8\",\"LILPQ\",\"OYSP7\",\"OYSP1\",\"CRPP5\",\"OSPP3\",\"OSPP11\",\"CRPP9\",\"CRPP2\",\"OSPP5\",\"CRPP4\",\"CRPPQ\",\"LILP5\",\"OSPP6\",\"OSPP7\",\"OSPP10\",\"OSPP2\",\"OSPP4\",\"OYSP3\",\"OYSP4\"]\n",
    "#fdir = \"/pine/scr/j/m/jmcgirr/pupfish_genomes/san_sal/raw_fastq/P/\"\n",
    "# S\n",
    "#infiles = [\"OSPS8\",\"OSPS2\",\"GRES4\",\"OYSS3\",\"GREP1\",\"OSPS11\",\"OSPS5\",\"LILS1\",\"GRES3\",\"OYSS5\",\"GREP2\",\"OSPS3\",\"LILS3\",\"LILS4\",\"MERP1\",\"LILS2\",\"OSPS7\",\"OSPS10\",\"OSPS1\",\"OSPS9\",\"OYSS4\",\"OYSS6\",\"MERP2\",\"OYSS9\",\"OSPS6\",\"OSPS4\"]\n",
    "#fdir = \"/pine/scr/j/m/jmcgirr/pupfish_genomes/san_sal/raw_fastq/S/\"\n",
    "# Carib\n",
    "infiles = [\"MEGQ1\",\"GEO2A8\",\"CATA1\",\"CUNP5\",\"BAVA8\",\"ARTA2\",\"GEO2A5\",\"VENA1\",\"CURA21\",\"VENA12\",\"GEO2A9\",\"NCCA11\",\"NCCA4\",\"GEOA2\",\"NCCA2\",\"NCCA5\",\"NCCA12\",\"CUNP7\",\"NCCA1\",\"NCCA15\",\"GEOA10\",\"BAVA11\",\"CUNA6\",\"NCCA9\",\"BAVA2\",\"FCTA1\",\"BAVA4\",\"VENA5\",\"CUNA1\",\"GEOA6\",\"VENA10\",\"MAY1\",\"BAVA6\",\"CUNP4\",\"VENA2\",\"BAVA5\",\"CUNA2\",\"BAVA10\",\"CUNA7\",\"VENA3\",\"GEO2A1\",\"CUNA10\",\"EXUA2\",\"NBIA1\",\"BAVA14\",\"CUNA4\",\"FLSA1\",\"CUNP11\",\"MAFA1\",\"VENA9\",\"CUNA3\",\"VENA7\",\"PWLA1\",\"ETA1\",\"GEOA7\",\"GEO2A6\",\"BAVA42\",\"BAVA7\",\"VENA8\",\"BAVA9\",\"GEO2A7\",\"NCCA8\",\"SIM1\",\"GEOA1\",\"ARTA1\",\"EPLA1\",\"NCCA3\",\"ACKA1\",\"GEOA5\",\"NCCA10\",\"VENA13\",\"GEOA4\",\"CUNA9\",\"KILA1\",\"GEOA11\",\"BAVA13\",\"VENA4\",\"CAIA1\",\"BAVA41\",\"CUNP3\",\"LGIA1\",\"EXUA1\",\"CUNA8\",\"DEAA1\",\"BAVA12\",\"CURA1\",\"CUNP6\",\"CURA2\",\"BONA1\",\"CUNA5\",\"NCCA7\",\"GEO2A3\",\"GEO2A10\",\"GEO2A4\",\"SCLA1\",\"GEO2A2\",\"SALA1\",\"NCCA6\"]\n",
    "fdir = \"/pine/scr/j/m/jmcgirr/pupfish_genomes/Caribbean_pups/\"\n",
    "# rna parents (already trimmed)\n",
    "#infiles = [\"CRPM1001\",\"CRPP1000\",\"CRPM1000\",\"NCCA1000\",\"CRPP1001\",\"CRPA1000\",\"OSPM1001\",\"CRPA1001\",\"CUNP10.2\",\"OSPA1001\",\"OSPP1000\",\"OSPA1000\",\"CRPA1003\",\"OSPM1000\",\"OSPP1001\"]\n",
    "\n",
    "for infile in infiles:\n",
    "\n",
    "    #sbatch submission script header\n",
    "    script = 'script_' + infile + job + '.sh'\n",
    "    outfile = io.open(script,'w', newline='\\n') \n",
    "     \n",
    "    outfile.write('#!/bin/bash\\n\\n#SBATCH --job-name='+infile+job+'\\n')\n",
    "    outfile.write('#SBATCH --mem=12G\\n')\n",
    "    outfile.write('#SBATCH --ntasks=4\\n')\n",
    "    outfile.write('#SBATCH --time=24:00:00 ## 1 week\\n')\n",
    "    outfile.write('#SBATCH -e '+infile+job+'_%A_%a.err ##error file if you want one (will be per job)\\n')\n",
    "    outfile.write('#SBATCH --mail-user=jmcgirr@email.unc.edu ##email you when job starts,ends,etc\\n#SBATCH --mail-type=ALL\\n\\n')\n",
    "  \n",
    "    #sbatch command\n",
    "    outfile.write('module load trim_galore \\n')\n",
    "    outfile.write('trim_galore -q 20 --paired --illumina '+fdir + infile +'_R1_001.fastq '+fdir + infile +'_R2_001.fastq \\n') \n",
    "    \n",
    "    #run sbatch submission \n",
    "    outfile.write('\\n\\n#command to run it on longleaf is sbatch '+script)\n",
    "    outfile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "job = '_align'\n",
    "main_dir = '/pine/scr/j/m/jmcgirr/pupfish_genomes/' \n",
    "trimmed_dir = main_dir + 'trimmed_fastqs/'\n",
    "sams_dir = main_dir + 'sams/'\n",
    "bams_dir = main_dir + 'bams/'\n",
    "variants_dir = main_dir + 'raw_variants/'\n",
    "\n",
    "#238 total\n",
    "infiles = [\"RHPA1\",\"SPPA1\",\"CRPA3\",\"MRKA1\",\"PIGA3\",\"WDPA1\",\"OSPA4\",\"ME2A2\",\"OSPA7\",\"OYSA1\",\"PIGA1\",\"OSPA6\",\"OSPA5\",\"CRPA1\",\"MERA2\",\"GNYA1\",\"LILA1\",\"ME2A1\",\"GREA1\",\"OSPA1\",\"OSPA9\",\"OYSA2\",\"CLRA1\",\"OSPA11\",\"NLLA1\",\"OSPA13\",\"GREA2\",\"OSPA12\",\"OSPA8\",\"PAIA1\",\"MERA3\",\"OSPA10\",\"OYSM8\",\"CRPM8\",\"CRPM7\",\"OSPM6\",\"OSPM7\",\"LILM5\",\"OYSM6\",\"OSPM5\",\"CRPM3\",\"CRPM6\",\"MRKM5\",\"OYSM1\",\"CRPM9\",\"CRPM2\",\"LILM4\",\"CRPM11\",\"OSPM11\",\"CRPM10\",\"OSPM3\",\"OSPM9\",\"OYSM3\",\"MRKM3\",\"OYSM2\",\"OSPM2\",\"OYSM5\",\"WDPM2\",\"LILMQ\",\"OSPM4\",\"OSPM10\",\"OSPM8\",\"LILM3\",\"OYSM4\",\"OSPM1\",\"OYSM7\",\"MRKM1\",\"MRKM4\",\"MRKM2\",\"CRPM1\",\"CRPM5\",\"ME2P1\",\"CRPP3\",\"CRPP7\",\"OSPP9\",\"LILP4\",\"CRPP8\",\"OSPP1\",\"LILP3\",\"OYSP6\",\"OSPP8\",\"LILPQ\",\"OYSP7\",\"OYSP1\",\"CRPP5\",\"OSPP3\",\"OSPP11\",\"CRPP9\",\"CRPP2\",\"OSPP5\",\"CRPP4\",\"CRPPQ\",\"LILP5\",\"OSPP6\",\"OSPP7\",\"OSPP10\",\"OSPP2\",\"OSPP4\",\"OYSP3\",\"OYSP4\",\"OSPS8\",\"OSPS2\",\"GRES4\",\"OYSS3\",\"GREP1\",\"OSPS11\",\"OSPS5\",\"LILS1\",\"GRES3\",\"OYSS5\",\"GREP2\",\"OSPS3\",\"LILS3\",\"LILS4\",\"MERP1\",\"LILS2\",\"OSPS10\",\"OSPS1\",\"OSPS9\",\"OYSS4\",\"OYSS6\",\"MERP2\",\"OYSS9\",\"OSPS6\",\"OSPS4\",\"MEGQ1\",\"GEO2A8\",\"CATA1\",\"CUNP5\",\"BAVA8\",\"ARTA2\",\"GEO2A5\",\"VENA1\",\"CURA21\",\"VENA12\",\"GEO2A9\",\"NCCA11\",\"NCCA4\",\"GEOA2\",\"NCCA2\",\"NCCA5\",\"NCCA12\",\"CUNP7\",\"NCCA1\",\"NCCA15\",\"GEOA10\",\"BAVA11\",\"CUNA6\",\"NCCA9\",\"BAVA2\",\"FCTA1\",\"BAVA4\",\"VENA5\",\"CUNA1\",\"GEOA6\",\"VENA10\",\"MAY1\",\"BAVA6\",\"CUNP4\",\"VENA2\",\"BAVA5\",\"CUNA2\",\"BAVA10\",\"CUNA7\",\"VENA3\",\"GEO2A1\",\"CUNA10\",\"EXUA2\",\"NBIA1\",\"BAVA14\",\"CUNA4\",\"FLSA1\",\"CUNP11\",\"MAFA1\",\"VENA9\",\"CUNA3\",\"VENA7\",\"PWLA1\",\"ETA1\",\"GEOA7\",\"GEO2A6\",\"BAVA42\",\"BAVA7\",\"VENA8\",\"BAVA9\",\"GEO2A7\",\"NCCA8\",\"SIM1\",\"GEOA1\",\"ARTA1\",\"EPLA1\",\"NCCA3\",\"ACKA1\",\"GEOA5\",\"NCCA10\",\"VENA13\",\"GEOA4\",\"CUNA9\",\"KILA1\",\"GEOA11\",\"BAVA13\",\"VENA4\",\"CAIA1\",\"BAVA41\",\"CUNP3\",\"LGIA1\",\"EXUA1\",\"CUNA8\",\"DEAA1\",\"BAVA12\",\"CURA1\",\"CUNP6\",\"CURA2\",\"BONA1\",\"CUNA5\",\"NCCA7\",\"GEO2A3\",\"GEO2A10\",\"GEO2A4\",\"SCLA1\",\"GEO2A2\",\"SALA1\",\"NCCA6\",\"CRPM1001\",\"CRPP1000\",\"CRPM1000\",\"NCCA1000\",\"CRPP1001\",\"CRPA1000\",\"OSPM1001\",\"CRPA1001\",\"CUNP10.2\",\"OSPA1001\",\"OSPP1000\",\"OSPA1000\",\"CRPA1003\",\"OSPM1000\",\"OSPP1001\"]\n",
    "\n",
    "\n",
    "for infile in infiles:\n",
    "\n",
    "    #sbatch submission script header\n",
    "    script = 'script_' + infile + job + '.sh'\n",
    "    outfile = io.open(script,'w', newline='\\n') \n",
    "     \n",
    "    outfile.write('#!/bin/bash\\n\\n#SBATCH --job-name='+infile+job+'\\n')\n",
    "    outfile.write('#SBATCH --mem=50G\\n')\n",
    "    outfile.write('#SBATCH --ntasks=4\\n')\n",
    "    outfile.write('#SBATCH --time=168:00:00 ## 1 week\\n')\n",
    "    outfile.write('#SBATCH -e '+infile+job+'_%A_%a.err ##error file if you want one (will be per job)\\n')\n",
    "    outfile.write('#SBATCH --mail-user=jmcgirr@email.unc.edu ##email you when job starts,ends,etc\\n#SBATCH --mail-type=ALL\\n\\n')\n",
    "    \n",
    "    outfile.write('module load bwa\\n')\n",
    "    outfile.write('module load samtools\\n')\n",
    "    outfile.write('bwa mem -aM -t 4 -R \"@RG\\\\tID:group1\\\\tSM:'+infile+'\\\\tPL:illumina\\\\tLB:lib1\" /proj/cmarlab/users/joe/Cyprinodon/bronto/asm.racon.fasta '+trimmed_dir+infile+'_R1_001_val_1.fq '+trimmed_dir+infile+'_R2_001_val_2.fq > '+sams_dir+infile+'.sam\\n')\n",
    "    outfile.write('module load samtools\\n')\n",
    "    outfile.write('samtools view -Shu '+sams_dir+infile+'.sam > '+bams_dir+infile+'.bam\\n')\n",
    "    outfile.write('samtools index '+bams_dir+infile+'.bam\\n')\n",
    "    outfile.write('rm '+sams_dir+infile+'.sam \\n')\n",
    "    outfile.write('samtools sort '+bams_dir+infile+'.bam -o '+bams_dir+infile+'.sort.bam\\n')\n",
    "    outfile.write('samtools index '+bams_dir+infile+'.sort.bam\\n')  \n",
    "\n",
    "    #run sbatch submission \n",
    "    outfile.write('\\n\\n#command to run it on longleaf is sbatch '+script)\n",
    "    outfile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "job = '_call_snps'\n",
    "main_dir = '/pine/scr/j/m/jmcgirr/pupfish_genomes/' \n",
    "bams_dir = main_dir + 'bams/sorted/'\n",
    "dedup_dir = main_dir + 'bams/dedup/'\n",
    "variants_dir = main_dir + 'raw_variants/'\n",
    "\n",
    "#238 total\n",
    "infiles = [\"RHPA1\",\"SPPA1\",\"CRPA3\",\"MRKA1\",\"PIGA3\",\"WDPA1\",\"OSPA4\",\"ME2A2\",\"OSPA7\",\"OYSA1\",\"PIGA1\",\"OSPA6\",\"OSPA5\",\"CRPA1\",\"MERA2\",\"GNYA1\",\"LILA1\",\"ME2A1\",\"GREA1\",\"OSPA1\",\"OSPA9\",\"OYSA2\",\"CLRA1\",\"OSPA11\",\"NLLA1\",\"OSPA13\",\"GREA2\",\"OSPA12\",\"OSPA8\",\"PAIA1\",\"MERA3\",\"OSPA10\",\"OYSM8\",\"CRPM8\",\"CRPM7\",\"OSPM6\",\"OSPM7\",\"LILM5\",\"OYSM6\",\"OSPM5\",\"CRPM3\",\"CRPM6\",\"MRKM5\",\"OYSM1\",\"CRPM9\",\"CRPM2\",\"LILM4\",\"CRPM11\",\"OSPM11\",\"CRPM10\",\"OSPM3\",\"OSPM9\",\"OYSM3\",\"MRKM3\",\"OYSM2\",\"OSPM2\",\"OYSM5\",\"WDPM2\",\"LILMQ\",\"OSPM4\",\"OSPM10\",\"OSPM8\",\"LILM3\",\"OYSM4\",\"OSPM1\",\"OYSM7\",\"MRKM1\",\"MRKM4\",\"MRKM2\",\"CRPM1\",\"CRPM5\",\"ME2P1\",\"CRPP3\",\"CRPP7\",\"OSPP9\",\"LILP4\",\"CRPP8\",\"OSPP1\",\"LILP3\",\"OYSP6\",\"OSPP8\",\"LILPQ\",\"OYSP7\",\"OYSP1\",\"CRPP5\",\"OSPP3\",\"OSPP11\",\"CRPP9\",\"CRPP2\",\"OSPP5\",\"CRPP4\",\"CRPPQ\",\"LILP5\",\"OSPP6\",\"OSPP7\",\"OSPP10\",\"OSPP2\",\"OSPP4\",\"OYSP3\",\"OYSP4\",\"OSPS8\",\"OSPS2\",\"GRES4\",\"OYSS3\",\"GREP1\",\"OSPS11\",\"OSPS5\",\"LILS1\",\"GRES3\",\"OYSS5\",\"GREP2\",\"OSPS3\",\"LILS3\",\"LILS4\",\"MERP1\",\"LILS2\",\"OSPS10\",\"OSPS1\",\"OSPS9\",\"OYSS4\",\"OYSS6\",\"MERP2\",\"OYSS9\",\"OSPS6\",\"OSPS4\",\"MEGQ1\",\"GEO2A8\",\"CATA1\",\"CUNP5\",\"BAVA8\",\"ARTA2\",\"GEO2A5\",\"VENA1\",\"CURA21\",\"VENA12\",\"GEO2A9\",\"NCCA11\",\"NCCA4\",\"GEOA2\",\"NCCA2\",\"NCCA5\",\"NCCA12\",\"CUNP7\",\"NCCA1\",\"NCCA15\",\"GEOA10\",\"BAVA11\",\"CUNA6\",\"NCCA9\",\"BAVA2\",\"FCTA1\",\"BAVA4\",\"VENA5\",\"CUNA1\",\"GEOA6\",\"VENA10\",\"MAY1\",\"BAVA6\",\"CUNP4\",\"VENA2\",\"BAVA5\",\"CUNA2\",\"BAVA10\",\"CUNA7\",\"VENA3\",\"GEO2A1\",\"CUNA10\",\"EXUA2\",\"NBIA1\",\"BAVA14\",\"CUNA4\",\"FLSA1\",\"CUNP11\",\"MAFA1\",\"VENA9\",\"CUNA3\",\"VENA7\",\"PWLA1\",\"ETA1\",\"GEOA7\",\"GEO2A6\",\"BAVA42\",\"BAVA7\",\"VENA8\",\"BAVA9\",\"GEO2A7\",\"NCCA8\",\"SIM1\",\"GEOA1\",\"ARTA1\",\"EPLA1\",\"NCCA3\",\"ACKA1\",\"GEOA5\",\"NCCA10\",\"VENA13\",\"GEOA4\",\"CUNA9\",\"KILA1\",\"GEOA11\",\"BAVA13\",\"VENA4\",\"CAIA1\",\"BAVA41\",\"CUNP3\",\"LGIA1\",\"EXUA1\",\"CUNA8\",\"DEAA1\",\"BAVA12\",\"CURA1\",\"CUNP6\",\"CURA2\",\"BONA1\",\"CUNA5\",\"NCCA7\",\"GEO2A3\",\"GEO2A10\",\"GEO2A4\",\"SCLA1\",\"GEO2A2\",\"SALA1\",\"NCCA6\",\"CRPM1001\",\"CRPP1000\",\"CRPM1000\",\"NCCA1000\",\"CRPP1001\",\"CRPA1000\",\"OSPM1001\",\"CRPA1001\",\"CUNP10.2\",\"OSPA1001\",\"OSPP1000\",\"OSPA1000\",\"CRPA1003\",\"OSPM1000\",\"OSPP1001\"]\n",
    "\n",
    "\n",
    "for infile in infiles:\n",
    "\n",
    "    #sbatch submission script header\n",
    "    script = 'script_' + infile + job + '.sh'\n",
    "    outfile = io.open(script,'w', newline='\\n') \n",
    "     \n",
    "    outfile.write('#!/bin/bash\\n\\n#SBATCH --job-name='+infile+job+'\\n')\n",
    "    outfile.write('#SBATCH --mem=50G\\n')\n",
    "    outfile.write('#SBATCH --ntasks=4\\n')\n",
    "    outfile.write('#SBATCH --time=168:00:00 ## 1 week\\n')\n",
    "    outfile.write('#SBATCH -e '+infile+job+'_%A_%a.err ##error file if you want one (will be per job)\\n')\n",
    "    outfile.write('#SBATCH --mail-user=jmcgirr@email.unc.edu ##email you when job starts,ends,etc\\n#SBATCH --mail-type=ALL\\n\\n')\n",
    "    \n",
    "    outfile.write('module load java\\n')\n",
    "    outfile.write('java -Xmx10g -jar /proj/cmarlab/users/joe/picard.jar MarkDuplicates INPUT='+bams_dir+infile+'.sort.bam OUTPUT='+dedup_dir+infile+'.sort.dedup.bam METRICS_FILE='+bams_dir+infile+'.metrics.txt MAX_FILE_HANDLES=1000 \\n')\n",
    "    outfile.write('module load samtools\\n')\n",
    "    outfile.write('samtools index '+dedup_dir+infile+'.sort.dedup.bam\\n')\n",
    "    outfile.write('module load gatk/3.8-0 \\n')\n",
    "    outfile.write('gatk -T HaplotypeCaller -ERC GVCF -drf DuplicateRead -R /proj/cmarlab/users/joe/Cyprinodon/bronto/asm.racon.fasta -I '+dedup_dir+infile+'.sort.dedup.bam -dontUseSoftClippedBases -stand_call_conf 20.0 -nct 4 -o '+variants_dir+infile+'_raw_variants.g.vcf')\n",
    "    \n",
    "    #run sbatch submission \n",
    "    outfile.write('\\n\\n#command to run it on longleaf is sbatch '+script)\n",
    "    outfile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "job = 'fst_all_v_m'\n",
    "fst_dir = '/pine/scr/j/m/jmcgirr/pupfish_genomes/popgen/'\n",
    "vcf = '/proj/cmarlab/users/joe/vcfs/all_pupfish_filtered_snps_passed.Q20.MAF0.05.MAXMISS0.5.recode.vcf'\n",
    "\n",
    "#sbatch submission script header\n",
    "script = 'script_' + job + '.sh'\n",
    "outfile = io.open(script,'w', newline='\\n')    \n",
    "outfile.write('#!/bin/bash\\n\\n#SBATCH --job-name='+job+'\\n')\n",
    "outfile.write('#SBATCH --mem=16G\\n')\n",
    "outfile.write('#SBATCH --ntasks=4\\n')\n",
    "outfile.write('#SBATCH --time=24:00:00 ##\\n')\n",
    "outfile.write('#SBATCH -e '+job+'_%A_%a.err ##error file if you want one (will be per job)\\n')\n",
    "outfile.write('#SBATCH --mail-user=jmcgirr@email.unc.edu ##email you when job starts,ends,etc\\n#SBATCH --mail-type=ALL\\n\\n')\n",
    "     \n",
    "#sbatch command\n",
    "outfile.write('module load vcftools \\n')\n",
    "outfile.write('vcftools --vcf '+vcf+' --keep '+fst_dir+'all.txt --out '+fst_dir+'all_v_m --weir-fst-pop '+fst_dir+'not_m.txt --weir-fst-pop '+fst_dir+'m.txt \\n')\n",
    "outfile.write('sed \\'s/-[0-9].*/0/g\\' '+fst_dir+'all_v_m.weir.fst  | sed \\'s/-nan/0/g\\' > '+fst_dir+'all_v_m_fst \\n')\n",
    "outfile.write('awk -F\\'\\\\t\\' \\'{ sum += $3 } END { print sum / NR }\\' '+fst_dir+'all_v_m_fst > '+fst_dir+'all_v_m_genome_wide_avg.txt \\n')\n",
    "\n",
    "#outfile.write('module load vcftools \\n')\n",
    "#outfile.write('vcftools --vcf '+vcf+' --keep '+fst_dir+'all.txt --out '+fst_dir+'all_v_p --weir-fst-pop '+fst_dir+'not_p.txt --weir-fst-pop '+fst_dir+'p.txt \\n')\n",
    "#outfile.write('sed \\'s/-[0-9].*/0/g\\' '+fst_dir+'all_v_p.weir.fst  | sed \\'s/-nan/0/g\\' > '+fst_dir+'all_v_p_fst \\n')\n",
    "#outfile.write('awk -F\\'\\\\t\\' \\'{ sum += $3 } END { print sum / NR }\\' '+fst_dir+'all_v_p_fst > '+fst_dir+'all_v_p_genome_wide_avg.txt \\n')\n",
    "\n",
    "#outfile.write('module load vcftools \\n')\n",
    "#outfile.write('vcftools --vcf '+vcf+' --keep '+fst_dir+'mp.txt --out '+fst_dir+'m_v_p --weir-fst-pop '+fst_dir+'m.txt --weir-fst-pop '+fst_dir+'p.txt \\n')\n",
    "#outfile.write('sed \\'s/-[0-9].*/0/g\\' '+fst_dir+'m_v_p.weir.fst  | sed \\'s/-nan/0/g\\' > '+fst_dir+'m_v_p_fst \\n')\n",
    "#outfile.write('awk -F\\'\\\\t\\' \\'{ sum += $3 } END { print sum / NR }\\' '+fst_dir+'m_v_p_fst > '+fst_dir+'m_v_p_genome_wide_avg.txt \\n')\n",
    "\n",
    "\n",
    "#run sbatch submission \n",
    "outfile.write('\\n')\n",
    "outfile.write('#command to run it on longleaf is sbatch '+script)\n",
    "\n",
    " \n",
    "outfile.close()\n",
    "\n",
    "# awk '{print $1}' all_v_m_fst > chr\n",
    "# awk '{print $2}' all_v_m_fst > ps\n",
    "# awk '{print $3}' all_v_m_fst > am\n",
    "# awk '{print $3}' all_v_p_fst > ap\n",
    "# awk '{print $3}' m_v_p_fst > mp\n",
    "# paste chr ps am ap mp > merged_fst.txt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "job = '_delly'\n",
    "main_dir = '/pine/scr/j/m/jmcgirr/pupfish_genomes/delly/' \n",
    "\n",
    "#238 total\n",
    "infiles = [\"RHPA1\",\"SPPA1\",\"CRPA3\",\"MRKA1\",\"PIGA3\",\"WDPA1\",\"OSPA4\",\"ME2A2\",\"OSPA7\",\"OYSA1\",\"PIGA1\",\"OSPA6\",\"OSPA5\",\"CRPA1\",\"MERA2\",\"GNYA1\",\"LILA1\",\"ME2A1\",\"GREA1\",\"OSPA1\",\"OSPA9\",\"OYSA2\",\"CLRA1\",\"OSPA11\",\"NLLA1\",\"OSPA13\",\"GREA2\",\"OSPA12\",\"OSPA8\",\"PAIA1\",\"MERA3\",\"OSPA10\",\"OYSM8\",\"CRPM8\",\"CRPM7\",\"OSPM6\",\"OSPM7\",\"LILM5\",\"OYSM6\",\"OSPM5\",\"CRPM3\",\"CRPM6\",\"MRKM5\",\"OYSM1\",\"CRPM9\",\"CRPM2\",\"LILM4\",\"CRPM11\",\"OSPM11\",\"CRPM10\",\"OSPM3\",\"OSPM9\",\"OYSM3\",\"MRKM3\",\"OYSM2\",\"OSPM2\",\"OYSM5\",\"WDPM2\",\"LILMQ\",\"OSPM4\",\"OSPM10\",\"OSPM8\",\"LILM3\",\"OYSM4\",\"OSPM1\",\"OYSM7\",\"MRKM1\",\"MRKM4\",\"MRKM2\",\"CRPM1\",\"CRPM5\",\"ME2P1\",\"CRPP3\",\"CRPP7\",\"OSPP9\",\"LILP4\",\"CRPP8\",\"OSPP1\",\"LILP3\",\"OYSP6\",\"OSPP8\",\"LILPQ\",\"OYSP7\",\"OYSP1\",\"CRPP5\",\"OSPP3\",\"OSPP11\",\"CRPP9\",\"CRPP2\",\"OSPP5\",\"CRPP4\",\"CRPPQ\",\"LILP5\",\"OSPP6\",\"OSPP7\",\"OSPP10\",\"OSPP2\",\"OSPP4\",\"OYSP3\",\"OYSP4\",\"OSPS8\",\"OSPS2\",\"GRES4\",\"OYSS3\",\"GREP1\",\"OSPS11\",\"OSPS5\",\"LILS1\",\"GRES3\",\"OYSS5\",\"GREP2\",\"OSPS3\",\"LILS3\",\"LILS4\",\"MERP1\",\"LILS2\",\"OSPS10\",\"OSPS1\",\"OSPS9\",\"OYSS4\",\"OYSS6\",\"MERP2\",\"OYSS9\",\"OSPS6\",\"OSPS4\",\"MEGQ1\",\"GEO2A8\",\"CATA1\",\"CUNP5\",\"BAVA8\",\"ARTA2\",\"GEO2A5\",\"VENA1\",\"CURA21\",\"VENA12\",\"GEO2A9\",\"NCCA11\",\"NCCA4\",\"GEOA2\",\"NCCA2\",\"NCCA5\",\"NCCA12\",\"CUNP7\",\"NCCA1\",\"NCCA15\",\"GEOA10\",\"BAVA11\",\"CUNA6\",\"NCCA9\",\"BAVA2\",\"FCTA1\",\"BAVA4\",\"VENA5\",\"CUNA1\",\"GEOA6\",\"VENA10\",\"MAY1\",\"BAVA6\",\"CUNP4\",\"VENA2\",\"BAVA5\",\"CUNA2\",\"BAVA10\",\"CUNA7\",\"VENA3\",\"GEO2A1\",\"CUNA10\",\"EXUA2\",\"NBIA1\",\"BAVA14\",\"CUNA4\",\"FLSA1\",\"CUNP11\",\"MAFA1\",\"VENA9\",\"CUNA3\",\"VENA7\",\"PWLA1\",\"ETA1\",\"GEOA7\",\"GEO2A6\",\"BAVA42\",\"BAVA7\",\"VENA8\",\"BAVA9\",\"GEO2A7\",\"NCCA8\",\"SIM1\",\"GEOA1\",\"ARTA1\",\"EPLA1\",\"NCCA3\",\"ACKA1\",\"GEOA5\",\"NCCA10\",\"VENA13\",\"GEOA4\",\"CUNA9\",\"KILA1\",\"GEOA11\",\"BAVA13\",\"VENA4\",\"CAIA1\",\"BAVA41\",\"CUNP3\",\"LGIA1\",\"EXUA1\",\"CUNA8\",\"DEAA1\",\"BAVA12\",\"CURA1\",\"CUNP6\",\"CURA2\",\"BONA1\",\"CUNA5\",\"NCCA7\",\"GEO2A3\",\"GEO2A10\",\"GEO2A4\",\"SCLA1\",\"GEO2A2\",\"SALA1\",\"NCCA6\",\"CRPM1001\",\"CRPP1000\",\"CRPM1000\",\"NCCA1000\",\"CRPP1001\",\"CRPA1000\",\"OSPM1001\",\"CRPA1001\",\"CUNP10.2\",\"OSPA1001\",\"OSPP1000\",\"OSPA1000\",\"CRPA1003\",\"OSPM1000\",\"OSPP1001\"]\n",
    "\n",
    "\n",
    "for infile in infiles:\n",
    "\n",
    "    #sbatch submission script header\n",
    "    script = 'script_' + infile + job + '.sh'\n",
    "    outfile = io.open(script,'w', newline='\\n') \n",
    "     \n",
    "    outfile.write('#!/bin/bash\\n\\n#SBATCH --job-name='+infile+job+'\\n')\n",
    "    outfile.write('#SBATCH --mem=50G\\n')\n",
    "    outfile.write('#SBATCH --ntasks=4\\n')\n",
    "    outfile.write('#SBATCH --time=168:00:00 ## 1 week\\n')\n",
    "    outfile.write('#SBATCH -e '+infile+job+'_%A_%a.err ##error file if you want one (will be per job)\\n')\n",
    "    outfile.write('#SBATCH --mail-user=jmcgirr@email.unc.edu ##email you when job starts,ends,etc\\n#SBATCH --mail-type=ALL\\n\\n')\n",
    "    \n",
    "    outfile.write('module load java\\n')\n",
    "    outfile.write('java -Xmx10g -jar /proj/cmarlab/users/joe/picard.jar MarkDuplicates INPUT='+bams_dir+infile+'.sort.bam OUTPUT='+dedup_dir+infile+'.sort.dedup.bam METRICS_FILE='+bams_dir+infile+'.metrics.txt MAX_FILE_HANDLES=1000 \\n')\n",
    "    outfile.write('module load samtools\\n')\n",
    "    outfile.write('samtools index '+dedup_dir+infile+'.sort.dedup.bam\\n')\n",
    "    outfile.write('module load gatk/3.8-0 \\n')\n",
    "    outfile.write('gatk -T HaplotypeCaller -ERC GVCF -drf DuplicateRead -R /proj/cmarlab/users/joe/Cyprinodon/bronto/asm.racon.fasta -I '+dedup_dir+infile+'.sort.dedup.bam -dontUseSoftClippedBases -stand_call_conf 20.0 -nct 4 -o '+variants_dir+infile+'_raw_variants.g.vcf')\n",
    "    \n",
    "    #run sbatch submission \n",
    "    outfile.write('\\n\\n#command to run it on longleaf is sbatch '+script)\n",
    "    outfile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "iters = np.arange(1,11)\n",
    "#gemma_dir = '/proj/cmarlab/users/joe/gemma/'\n",
    "gemma_dir = '/pine/scr/j/m/jmcgirr/gemma_emilie/'\n",
    "# -n 2 = pigment 3 = low jaw 4 = nose length 5 = nose height\n",
    "\n",
    "for i in iters:\n",
    "    job = 'run_gemma_nose_height'\n",
    "    j = str(i)\n",
    "    #sbatch submission script header\n",
    "    script = 'script_' + job + '_' + j + '.sh'\n",
    "    outfile = io.open(script,'w', newline='\\n') \n",
    "    jobname= 'script_' + job +j\n",
    "     \n",
    "    outfile.write('#!/bin/bash\\n\\n')\n",
    "    outfile.write('#SBATCH --job-name='+jobname+'\\n')\n",
    "    outfile.write('#SBATCH --mem=16G\\n')\n",
    "    outfile.write('#SBATCH --ntasks=4\\n')\n",
    "    outfile.write('#SBATCH --time=72:00:00 ##\\n')\n",
    "    outfile.write('#SBATCH -e '+jobname+'_%A_%a.err ##error file if you want one (will be per job)\\n')\n",
    "    outfile.write('#SBATCH --mail-user=jmcgirr@email.unc.edu ##email you when job starts,ends,etc\\n')\n",
    "    outfile.write('#SBATCH --mail-type=ALL\\n')\n",
    "    outfile.write('\\n')\n",
    "    \n",
    "    #sbatch command\n",
    "    outfile.write('/proj/cmarlab/users/joe/gemma/gemma-0.98.1-linux-static -bfile '+gemma_dir+'all_pupfish_filtered_snps_passed.Q20.MAF0.05.MAXMISS0.5.plink -w 50000000 -s 100000000 -n 5 -rpace 10000 -wpace 100000 -bslmm 1 -o all_pupfish_filtered_snps_passed.Q20.MAF0.05.MAXMISS0.5.plink_sansal_run_'+j+' \\n')\n",
    "    #outfile.write(gemma_dir +'gemma-0.98.1-linux-static -bfile '+gemma_dir+'58_dna_maf_0.5_maxmiss_0.9_snps.recode.vcf.plink -w 50000000 -s 100000000 -n 3 -rpace 10000 -wpace 100000 -bslmm 1 -o gemma_low_jaw_'+j+' \\n')\n",
    "    #outfile.write(gemma_dir +'gemma-0.98.1-linux-static -bfile '+gemma_dir+'58_dna_maf_0.5_maxmiss_0.9_snps.recode.vcf.plink -w 50000000 -s 100000000 -n 2 -rpace 10000 -wpace 100000 -bslmm 1 -o gemma_up_jaw_'+j+' \\n')\n",
    "\n",
    "    #run sbatch submission \n",
    "    outfile.write('\\n')\n",
    "    outfile.write('#command to run it on longleaf is sbatch '+script)\n",
    "    \n",
    "     \n",
    "    outfile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# create fasta consensus for inversion regions\n",
    "job = 'consensus_fasta'\n",
    "\n",
    "scaff = 'HiC_scaffold_46'\n",
    "start = '32757963'\n",
    "stop  = '32793397'\n",
    "coords = scaff+':'+start+'-'+stop\n",
    "\n",
    "bam_dir = \"/pine/scr/j/m/jmcgirr/pupfish_genomes/bams/dedup/\"\n",
    "infiles = [\"CRPP1001\"]\n",
    "infiles = [\"OSPP1001\",\"OSPP1000\",\"OSPM1001\",\"OSPM1000\",\"CRPP1001\",\"CRPP1000\",\"CRPM1001\",\"CRPM1000\"]\n",
    "\n",
    "for infile in infiles:\n",
    "\n",
    "    #sbatch submission script header\n",
    "    script = 'script_' + infile + job + '.sh'\n",
    "    outfile = io.open(script,'w', newline='\\n') \n",
    "     \n",
    "    outfile.write('#!/bin/bash\\n\\n#SBATCH --job-name='+infile+job+'\\n')\n",
    "    outfile.write('#SBATCH --mem=8G\\n')\n",
    "    outfile.write('#SBATCH --ntasks=4\\n')\n",
    "    outfile.write('#SBATCH --time=24:00:00 ## 1 week\\n')\n",
    "    outfile.write('#SBATCH -e '+infile+job+'_%A_%a.err ##error file if you want one (will be per job)\\n')\n",
    "    outfile.write('#SBATCH --mail-user=jmcgirr@email.unc.edu ##email you when job starts,ends,etc\\n#SBATCH --mail-type=ALL\\n\\n')\n",
    "    \n",
    "    outfile.write('module load samtools \\n')\n",
    "    outfile.write('#samtools mpileup -g -f /proj/cmarlab/users/joe/Cyprinodon/bronto/asm.racon.fasta '+bam_dir+infile+'.sort.dedup.bam > /pine/scr/j/m/jmcgirr/pupfish_genomes/inversions/'+infile+'.bcf\\n')\n",
    "    outfile.write('#bcftools call --threads 4 -m /pine/scr/j/m/jmcgirr/pupfish_genomes/inversions/'+infile+'.bcf > /pine/scr/j/m/jmcgirr/pupfish_genomes/inversions/'+infile+'.vcf\\n')\n",
    "    outfile.write('#bgzip -c /pine/scr/j/m/jmcgirr/pupfish_genomes/inversions/'+infile+'.vcf > /pine/scr/j/m/jmcgirr/pupfish_genomes/inversions/'+infile+'.vcf.gz\\n')\n",
    "    outfile.write('#tabix /pine/scr/j/m/jmcgirr/pupfish_genomes/inversions/'+infile+'.vcf.gz\\n')\n",
    "    outfile.write('samtools faidx /proj/cmarlab/users/joe/Cyprinodon/bronto/asm.racon.fasta '+coords+' | bcftools consensus /pine/scr/j/m/jmcgirr/pupfish_genomes/inversions/'+infile+'.vcf.gz > /pine/scr/j/m/jmcgirr/pupfish_genomes/inversions/'+infile+'.fa\\n')\n",
    "\n",
    "    #run sbatch submission \n",
    "    outfile.write('\\n\\n#command to run it on longleaf is sbatch '+script)\n",
    "    outfile.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "job = 'indels'\n",
    "vcf_dir= '/proj/cmarlab/users/joe/vcfs/'\n",
    "\n",
    "#sbatch submission script header\n",
    "script = 'script_' + job + '.sh'\n",
    "outfile = io.open(script,'w', newline='\\n')    \n",
    "outfile.write('#!/bin/bash\\n\\n#SBATCH --job-name='+job+'\\n')\n",
    "outfile.write('#SBATCH --mem=32G\\n')\n",
    "outfile.write('#SBATCH --ntasks=4\\n')\n",
    "outfile.write('#SBATCH --time=168:00:00 ##\\n')\n",
    "outfile.write('#SBATCH -e '+job+'_%A_%a.err ##error file if you want one (will be per job)\\n')\n",
    "outfile.write('#SBATCH --mail-user=jmcgirr@email.unc.edu ##email you when job starts,ends,etc\\n#SBATCH --mail-type=ALL\\n\\n')\n",
    "     \n",
    "#sbatch command\n",
    "outfile.write('module load vcftools \\n')\n",
    "outfile.write('vcftools --gzvcf '+vcf_dir+'pupfish_february_2019_assembly.vcf.gz --keep-only-indels --max-missing 0.5 --minQ 20 --recode --recode-INFO-all --out pupfish_february_2019_indels \\n')    \n",
    "\n",
    "#run sbatch submission \n",
    "outfile.write('\\n')\n",
    "outfile.write('#command to run it on longleaf is sbatch '+script)\n",
    "\n",
    " \n",
    "outfile.close()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
